physionet-django/project/views.py

```diff
@@ -2294,7 +2294,7 @@ def generate_signed_url(request, project_slug):
         return JsonResponse({'detail': 'The file size cannot be a negative value.'}, status=400)
 
     try:
-        validate_filename(filename)
+        validate_gcs_bucket_object_name(filename)
```

Comment:
I haven't looked at this PR in detail, so I might be missing something, but it looks like this change means that file validation will now vary between platforms, depending on whether or not they are using the `generate_signed_url` method? 

If possible, I think it would be best to keep all platforms in sync, using the strictest validation rule in all cases. This helps to maintain consistent platforms and would help to ensure it is possible to switch between approaches over time.

Reply:
That is correct @tompollard  The reason for validation being separate here is that gcs allows file names with `/` in the names(i think slashes are used to determine directory). For example if we upload files with name `folder1/folder2/file.txt`, then the file.txt will be uploaded inside the folder1/folder2 directory.

For file validation during upload in project, i think there are two methods 1)gcs 2) upload to the server directly.

for the method (2), i think the same validation is done on the forms(except the / part)


@kshalot  might be able to provider a better explanation

Reply:
The GCS upload is different because of the fact that `subdir/file` is not a file inside a subdirectory, but a single object which's name is literally `subdir/file` (so there is no such thing as a directory). So the underlying mechanisms imply a slight different way to validate them. Having said that, I think what @tompollard said has value - we can make the GCS validation overlap with the regular filename validation as much as possible to make potential migrations easy.

Since both directories and files in the local storage type are validated with `validate_filename`, we could implement `validate_gcs_bucket_object_name` as:
```python
# The full length still has to be checked since `validate_filename` only checks the length of a single segment
if len(value) > MAX_GCS_OBJECT_NAME_LENGTH:
  raise ValidationError(...)

for path_segment in filename.split('/'):
  validate_filename(path_segment)
```

This way the GCS backend should be compatible with the local one, as it's stricter (enforces shorter path in project directory).

Reply:
Thats a good idea @kshalot . Thanks

@tompollard , i will make the changes and push them. 

@tompollard  would you like me to squash the changes? I am a bit torn between keeping the record of this decision in git history, vs  not keeping it.

Reply:
@tompollard  i pushed the change  with 1 extra commit message, Will squash the last two commits if that is better.

Please let me know